lib/readline.js

```diff
@@ -47,9 +47,9 @@ function Interface(input, output, completer, terminal) {
   }
   historySize = historySize || kHistorySize;
 
-  completer = completer || function() { return []; };
+  completer = completer || undefined;
 
-  if (typeof completer !== 'function') {
+  if (! typeof completer in ['function', 'undefined']) {
```

Comment:
I just ran my own benchmark (which, hopefully, I didn't mess up)n and `(typeof completer !== 'function' && typeof completer !==  'undefined')` is more than three times as fast as `(['function', 'undefined'].indexOf(typeof completer) === -1)`. So I guess I need to (a) add a test and (b) change the formulation to yours.